bundle/src/main/java/com/adobe/acs/commons/wcm/impl/PropertyMergePostProcessor.java

```diff
@@ -21,183 +21,257 @@
+     * property as a multi-value. The values of the source properties and
+     * destination properties must all be the same property type.
      *
      * The unique set of properties will be stored in
      *
-     * @param resource    the resource to look for the source and destination properties on
+     * @param resource the resource to look for the source and destination
+     * properties on
      * @param destination the property to store the collected properties.
-     * @param sources  the properties to collect values from for merging
-     * @param typeHint the data type that should be used when reading and storing the data
-     * @param allowDuplicates true to allow duplicates values in the destination property; false to make values unique
-     * @return true if changes were made to the destination property
+     * @param sources the properties to collect values from for merging
+     * @param typeHint the data type that should be used when reading and
+     * storing the data
+     * @param allowDuplicates true to allow duplicates values in the destination
+     * property; false to make values unique
+     * @return Optional resource updated, if any
      */
-    protected final <T> boolean merge(final Resource resource, final String destination,
-                                      final List<String> sources, final Class<T> typeHint,
-                                      final boolean allowDuplicates) throws PersistenceException {
+    protected final <T> Optional<Resource> merge(final Resource resource, final String destination,
+            final Collection<String> sources, final Class<T> typeHint,
+            final boolean allowDuplicates) throws PersistenceException {
 
         // Create an empty array of type T
         @SuppressWarnings("unchecked")
         final T[] emptyArray = (T[]) Array.newInstance(typeHint, 0);
 
-        final ModifiableValueMap properties = resource.adaptTo(ModifiableValueMap.class);
+        final String targetProperty;
 
         Collection<T> collectedValues = null;
 
         if (allowDuplicates) {
-            collectedValues = new ArrayList<T>();
+            collectedValues = new ArrayList<>();
         } else {
-            collectedValues = new LinkedHashSet<T>();
+            collectedValues = new LinkedHashSet<>();
         }
 
         for (final String source : sources) {
-            // Get the source value as type T
-            final T[] tmp = properties.get(source, emptyArray);
-
-            // If the value is not null, add to collectedValues
-            if (tmp != null) {
-                collectedValues.addAll(Arrays.asList(tmp));
+            Resource sourceProperties = resource;
+            String sourceParam = source;
+            if (source.contains("/")) {
+                sourceProperties = resource.getResourceResolver().getResource(resource, source.substring(0, source.lastIndexOf('/')));
```

Comment:
`StringUtils.substringAfterLast("source", "/")` is a bit easier to read.

also forgiving if there is not "/" at all.. just returns the String. 

https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html#substringAfterLast-java.lang.String-java.lang.String-

Reply:
Oh nice!  I didn't know about that method.  Thanks!